gosushi_server/util/calculatePoints.js

```diff
@@ -14,61 +14,56 @@ const addPlayerPoints = (playerName, players, points) => {
 
 calculateTurnPoints = (players, menu, uramakiCountMap, uramakiStanding) => {
   players.forEach(currPlyr => {
+    // Get list of unused wasabi cards
     let wasabi = currPlyr.playedCards.filter(
       card => card.name === cardNameEnum.WASABI && !card.data
     );
-    let tripleNigiri = false;
-    if (menu.specials.includes(cardNameEnum.WASABI) && wasabi.length > 0) {
-      tripleNigiri = true;
-    }
     while (currPlyr.turnCards.length !== 0) {
       const card = currPlyr.turnCards.pop();
       switch (card.name) {
         case cardNameEnum.EGG:
-          if (tripleNigiri) {
+          // Update wasabi with score to triple the nigiri's value
+          if (wasabi.length > 0) {
             wasabi[0].data = 2;
-            wasabi = currPlyr.playedCards.filter(
-              card => card.name === cardNameEnum.WASABI && !card.data
-            );
-            tripleNigiri = wasabi.length > 0;
+            // Shift the used wasabi out of the list of unused wasabis
+            wasabi.shift();
           }
           break;
         case cardNameEnum.SALMON:
-          if (tripleNigiri) {
+          if (wasabi.length > 0) {
             wasabi[0].data = 4;
-            wasabi = currPlyr.playedCards.filter(
-              card => card.name === cardNameEnum.WASABI && !card.data
-            );
-            tripleNigiri = wasabi.length > 0;
+            wasabi.shift();
           }
           break;
         case cardNameEnum.SQUID:
-          if (tripleNigiri) {
+          if (wasabi.length > 0) {
             wasabi[0].data = 6;
-            wasabi = currPlyr.playedCards.filter(
-              card => card.name === cardNameEnum.WASABI && !card.data
-            );
-            tripleNigiri = wasabi.length > 0;
+            wasabi.shift();
           }
           break;
         case cardNameEnum.MISO_SOUP:
+          // Find all other players that played a miso
           const repeats = players.filter(
             otherPlyr =>
               otherPlyr.name !== currPlyr.name &&
               otherPlyr.turnCards.find(c => c.name === cardNameEnum.MISO_SOUP)
```

Comment:
This is fine but we're going to have to remember to add some 'no duplicate player names' validation on join game

Reply:
Oh and also I'm aware this^ is from previous commits its just something to think about in general